### DISCUSSION TITLE
Could novelWriter be used to write a "how to" book?

### DISCUSSION THREAD

**Discussion by Wysardry** [vkbo/novelWriter#2026] [Q&A] [answered]
The documentation says that novelWriter is not suitable for technical writing, but does not go into much detail as to _why_ that is.

I was thinking of using it to create a "how to" book, such as "how to make wine" or "how to prune a fruit tree".

Would novelWriter work for this?

Also, am I right in thinking that novelWriter could be installed on a Raspberry Pi 4 using pip?

**Comment by vkbo** [comment]
It is unsuitable for technical writing in the sense that it is plain text with some minimal formatting support for emphasis and a few other simple formatting features. If that's all you need, then sure. It can be used.

As for Raspberry Pi, if the dependencies can be installed (Python, PyQt5 and PyEnchant) then novelWriter itself is just Python code and will in principle run on anything where these three dependencies can be installed. If the file system is quirky, then there may be issues with that, but aside from this, I don't know of any limitations.

**Reply by Wysardry** [reply]
I would probably need to use markdown codes for images, links and definition lists, but I would be okay with them not being shown in a WYSIWYG form.

I could always use a dedicated markdown editor to check everything displays correctly at regular intervals.

The ability to split the text into multiple sections that can be moved around is more important to me than the preview feature.

AFAIK, the main differences between a Raspberry Pi and a standard Linux machine are the ARM CPU, the relatively small RAM and the use of SD cards for storage. Mine has 4Gb of RAM, which I'm guessing should be more than enough to run novelWriter as Zettlr only needs 1Gb.

**Reply by vkbo** [reply, marked as answer]
> I would probably need to use markdown codes for images, links and definition lists, but I would be okay with them not being shown in a WYSIWYG form.

Well, novelWriter would ignore lists, images and links, sure. You can also build the text manuscript in both standard Markdown (which will parse the syntax the editor does understand) and as txt files in the native novelWriter syntax which will not process most of it. All it does is pre-process headings if you have special heading formatting options enabled, which by default are set to not modify headings.

> AFAIK, the main differences between a Raspberry Pi and a standard Linux machine are the ARM CPU, the relatively small RAM and the use of SD cards for storage. Mine has 4Gb of RAM, which I'm guessing should be more than enough to run novelWriter as Zettlr only needs 1Gb.

SD cards are not an issue, and novelWriter doesn't use much RAM on its own, but the Qt framework uses a bit. Still, we're talking on the order of 100MB or so if I remember correctly.

You can track memory usage by starting novelWriter from command line with the `--meminfo` flag and passing the memory tracking env variable `PYTHONTRACEMALLOC` used by Python's [tracemalloc](https://docs.python.org/3/library/tracemalloc.html) module. The `--meminfo` flag in novelWriter will display the info in the status bar.
